pkg/storage/stores/shipper/compactor/retention/expiration.go

```diff
@@ -119,17 +132,19 @@ Outer:
 	return globalRetention
 }
 
-// findLatestRetentionStartTime returns the latest retention start time overall and by each user.
-func findLatestRetentionStartTime(now model.Time, limits Limits) (model.Time, map[string]model.Time) {
+// findLatestRetentionStartTime returns the latest retention start time overall, just default config and by each user.
+func findLatestRetentionStartTime(now model.Time, limits Limits) (model.Time, model.Time, map[string]model.Time) {
```

Comment:
nit: I'm happy with the return here. I think in the future we should make that a struct with proper names, this would avoid confusion.

Reply:
makes sense, pushed this change